documentation/SolutionsRegistryFormat.md

```diff
@@ -143,6 +143,37 @@ This directive is used to detect whether a solution is installed. If any of thes
 ]
 ```
 
+####isConfigurable
+This is run before configuration to ensure that the application is actually ready to be configured. This is relevant for applications where e.g. a configuration file needs to be present, a wizard needs to be run on the first launch, etc. Each of the blocks will be evaluated, if *any* of them evaluates to true, the solution is considered configurable. The absence of an `isConfigurable` block is also interpreted as the solution being configurable.
+
+If a solution is not configurable, the `makeConfigurable` block will be executed immediately following the `isConfigurable` check (see below).
+
+**Example Entry**:
+```
+"isConfigurable": [
+    {
+        "type": "gpii.lifecycleActions.pathExists",
+        "path": "/tmp/fakemag1.settings.json"
+    }
+]
+```
+
+####makeConfigurable
+Is the actions that need to be taken to make the application configurable (such as running a wizard, creating a default configuration file, adding a new system user, etc). This will be triggered by the system if `isConfigurable` has evaluated to false and run immediately following the `isConfigurable` check.
+
+**Example Entry**:
+```
+"makeConfigurable": [
+    {
+        "type": "gpii.lifecycleActions.createFile",
```

Comment:
Yes, I can confirm that behavior with both Orca and NVDA. Isn't it a good use case? 
Can't we just move my implementation of "makeConfigurable" (https://github.com/GPII/linux/blob/a0acb4730edd0382dfd2010a9a713cff29268523/gpii/node_modules/orca/orcaSettingsHandler.js#L118) into a utility function in universal that can be used by any application that requires a start/stop cycle before being configured? By having this and by adding a few improvements to the JSON settings handler (mostly, being able to write the settings in the right profile), the infamous orca settings handler would disappear for the rest of the days :dancer: